services/analyzer/custom_rules.py

```
@@ -0,0 +1,64 @@
+                config = yaml.safe_load(file)
+                if config and 'rules' in config:
+                    self.rules = config['rules']
+                    for rule in self.rules:
+                        try:
+                            rule['compiled_pattern'] = re.compile(rule['pattern'])
+                        except re.error as e:
+                            logger.error(f"Invalid regex pattern in rule '{rule.get('name')}': {e}")
+                            rule['compiled_pattern'] = None
+                    
+                    logger.info(f"Successfully loaded {len(self.rules)} custom security rules.")
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)